etcdserver/apply.go

```diff
@@ -745,3 +745,10 @@ func compareInt64(a, b int64) int {
 func isGteRange(rangeEnd []byte) bool {
 	return len(rangeEnd) == 1 && rangeEnd[0] == 0
 }
+
+func noSideEffect(r *pb.InternalRaftRequest) bool {
+	if r.Txn != nil {
+		return isTxnReadonly(r.Txn)
```

Comment:
probably we can do the txn one later inside the actual txn apply func. we only count range, authget as noSideEffect here.